# Assignment 6

To measure the psychological state of Americans, a psychology researcher (in 1929) devised
three questions put to 3443 Americans from different walks of life. He asked them whether they
were happy. They could answer only **yes** or **no** to each question. By today’s standards the study
is quite simple, and he did no formal statistical analysis of his data. The frequency data (number
of **yes** and **no** responses within each employment category) from this study are in the file
*Americandata.csv*.

---

You need to use the data and answer the question, that is, **Is the type of profession related to the happiness of American people?**
Report the **χ2** statistic and the *statistical significance of test*, in addition to the contingency table.
Also, explain the results and what they mean. What does the value of **χ2** statistic mean? Is it
good or bad? Is the test statistically significant?

In [58]:
import pandas as pd
import scipy.stats as stats

#inspect data 
df = pd.read_csv("./Americandata.csv")
df.head(10)


,Happy,Students,Laborers,Preachers,Physicians,Housewives,Teachers,Lawyers,Musicians
0,Yes,390,378,35,159,78,108,11,31
1,No,1610,122,265,51,122,38,64,19


In [136]:
# Contingency tables
def contingency_table (name):
    #create contingency table 
    table = pd.crosstab (df["Happy"], df[name])
    print(name.upper())
    print(f"Contingency table:\n {table}")

    #run chi-square testing
    chi2, p_value, dof, expected = stats.chi2_contingency(table, correction=False)
    print(f"\nChi-square test: \nChi-square: {chi2}\np_value: {p_value:.6f}\ndegrees of freedom: {dof}\n")

    #return expected outcomes 
    print("Expected counts- if variables were UNRELATED:")
    expected_df = pd.DataFrame(expected, index = table.index, columns = table.columns)
    print(expected_df.round(2))
    #checking assumption 
    if (expected >=5).all():
        print("Assumption okay; all counts greater than 5.")
    else:
        print("WARNING! some expected counts may be less than 5.")

    print()

    #critical values decison
    alpha = 0.05
    critical_value = stats.chi2.ppf(1 - alpha, dof)
    print(f"Critical value (alpha = {alpha}. Degree of freedom = {dof}): {critical_value:.2f}")
    print(F"Our Chi-square= {chi2}")


    #p-value decision
    if chi2 > critical_value:
        print(f"Since {chi2:.2f} > {critical_value:.2f}, Reject independence (reject the null hypothesis) -> role and hero ARE related")
    else:
        print(f"Since {chi2:.2f} <= {critical_value:.2f}, Cannot reject") #you accept null, there is no relation

    print()

    if p_value <= alpha: 
        print(f"{p_value:6f} < {alpha}: Reject null hypothesis- happiness and profession are related \n(dependent)")
    else: 
        print(f"{p_value:6f} >= {alpha}: Fail to reject null hypothesis- there is not enough evidence that happiness and profession are related\n(independent)")

    print("="*120)
    print()


contingency_table("Students")
contingency_table("Laborers")
contingency_table("Preachers")
contingency_table("Physicians")
contingency_table("Housewives")
contingency_table("Teachers")
contingency_table("Lawyers")
contingency_table("Musicians")

STUDENTS
Contingency table:
 Students  390   1610
Happy               
No           0     1
Yes          1     0

Chi-square test: 
Chi-square: 2.0
p_value: 0.157299
degrees of freedom: 1

Expected counts- if variables were UNRELATED:
Students  390   1610
Happy               
No         0.5   0.5
Yes        0.5   0.5
WARNING! some expected counts may be less than 5.

Critical value (alpha = 0.05. Degree of freedom = 1): 3.84
Our Chi-square= 2.0
Since 2.00 <= 3.84, Cannot reject

0.157299 >= 0.05: Fail to reject null hypothesis- there is not enough evidence that happiness and profession are related
(independent)

LABORERS
Contingency table:
 Laborers  122  378
Happy             
No          1    0
Yes         0    1

Chi-square test: 
Chi-square: 2.0
p_value: 0.157299
degrees of freedom: 1

Expected counts- if variables were UNRELATED:
Laborers  122  378
Happy             
No        0.5  0.5
Yes       0.5  0.5
WARNING! some expected counts may be less than 5.

Critical value (alpha = 0.

You need to use the data and answer the question, that is, **Is the type of profession related to the happiness of American people?**
Report the **χ2** statistic and the *statistical significance of test*, in addition to the contingency table.
Also, explain the results and what they mean. What does the value of **χ2** statistic mean? Is it
good or bad? Is the test statistically significant?

Put your analysis in the empty `markdown` cell below.

I ran the test thorough every profession- I am not sure if there was a more efficient way to do this, but after creating the contingency tables, it was clear that all the results would be the same. 
<br>
I ran the chi-square test on the data, which measures the observed data to the expected data and compares the frequencies of selections (how often professionals chose yes or no). As we saw in the contingency table, the frequency at which yes or no were selected were roughly equal across all professions. The χ2 statstic is 2.0, which is less than the critical value (3.84). The critical value is used to see if the test is strong enough to reject the null, and since 2 < ~3, we cannot reject the null hypothesis. This is further supported with the p value calculations. The p-value is the probability of results occuring by chance, and the lower p value means there is more evidence against the null hypothesis. Since the p value 0.157 > 0.05, this means that we fail to reject the null as there is not enough evidence to suggest that happiness and profession are related. It is likely that these results happened by chance. Therefore, the type of profession likely does not relate to the happiness of the American people. However, this test and analysis requires more data and has to take in account other factors to be more reliable- measuring happiness and profession is a very narrow scope and limits the validity of results. 